oedipus/__init__.py

```
@@ -219,6 +251,28 @@ class S(object):
 
     __len__ = count
 
+    def excerpt(self, result):
+        """Takes a result and returns the excerpt as a string."""
+        if self._results_cache is None:
+            # This guarantees that we have results before we try
+            # getting excerpts for results.
+            self._raw()
```

It's kind of non-trivial, but you can kick up a KeyError if you call excerpt on an S with results from another S.  One shouldn't be doing that sort of thing, but getting a KeyError would be puzzling.  I figured it's an easy/fast check to see if we've already got results or not.

Having said that, this is all kind of weird and if you're in this situation, it's better to throw an error rather than let it slide through and get an excerpt with settings from a different S.  So maybe instead of calling self._raw(), it's better to throw a DudeThisSHasNoResultsYet error.  What do you think?